cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.system;
+
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.data2.metadata.store.NoOpMetadataStore;
+import co.cask.cdap.proto.Id;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test DatasetSystemMetadataWriter
+ */
+public class DatasetSystemMetadataWriterTest {
+  @Test
+  public void testFilesetSchema() throws Exception {
+    DatasetProperties filesetAvroTableProps =
+      FileSetProperties.builder()
+        .setTableProperty("avro.schema.literal", "avro-table-schema")
+        .build();
+    DatasetSystemMetadataWriter metadataWriter =
+      new DatasetSystemMetadataWriter(new NoOpMetadataStore(), Id.DatasetInstance.from("ns1", "avro1"),
+                                      filesetAvroTableProps, null, null, null);
+    Assert.assertEquals("avro-table-schema", metadataWriter.getSchemaToAdd());
+
+    // When SCHEMA property is present, it should override
+    filesetAvroTableProps =
+      FileSetProperties.builder()
+        .setTableProperty("avro.schema.literal", "avro-table-schema")
+        .add(DatasetProperties.SCHEMA, "avro-schema")
+        .build();
+    metadataWriter =
+      new DatasetSystemMetadataWriter(new NoOpMetadataStore(), Id.DatasetInstance.from("ns1", "avro1"),
+                                      filesetAvroTableProps, null, null, null);
+    Assert.assertEquals("avro-schema", metadataWriter.getSchemaToAdd());
+
+    DatasetProperties filesetAvroOutputProps =
```

Comment:
nit: can these verifications be easily extracted to methods? seems like a similar code.